src/main/java/com/exasol/releasedroid/adapter/communityportal/CommunityPost.java

```diff
@@ -0,0 +1,158 @@
+
+import java.security.SecureRandom;
+import java.util.List;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * This class represents a post on the Exasol Community portal.
+ */
+public class CommunityPost {
+    private static final List<String> TEASERS = List.of("See what's new here.", "Learn more here.",
+            "Find out what's new here.", "Find out how it can help you.",
+            "Find out what's changed and where you can learn more.", "Read more here.", "Find out more here.",
+            "Find out what this means, here.");
+    private final SecureRandom random = new SecureRandom();
+    private final String header;
+    private final String body;
+    private final String boardId;
+    private final String teaser;
+    private final List<String> tags;
+
+    private CommunityPost(final Builder builder) {
+        this.header = builder.header;
+        this.body = builder.body;
+        this.boardId = builder.boardId;
+        this.tags = builder.tags;
+        this.teaser = builder.teaser;
+    }
+
+    /**
+     * Get the post's header.
+     * 
+     * @return post's header
+     */
+    public String getHeader() {
+        return this.header;
+    }
+
+    /**
+     * Get the post's body.
+     * 
+     * @return post's body
+     */
+    public String getBody() {
+        return this.body;
+    }
+
+    /**
+     * Get an id of the board to add the post to.
+     * 
+     * @return id of the board
+     */
+    public String getBoardId() {
+        return this.boardId;
+    }
+
+    /**
+     * Get a list of tags of the post.
+     * 
+     * @return list of tags of the post
+     */
+    public List<String> getTags() {
+        return this.tags;
+    }
+
+    /**
+     * Get a teaser for an article. If the teaser was not set, it returns a random teaser.
+     * 
+     * @return teaser
+     */
+    public String getTeaser() {
+        if (this.teaser == null || this.teaser.isEmpty()) {
+            return TEASERS.get(this.random.nextInt(TEASERS.size()));
+        } else {
+            return this.teaser;
+        }
+    }
+
+    /**
+     * Transform the community post to the JSON format.
+     * 
+     * @return community post as a JSON string
+     */
+    public String toJson() {
+        final var board = new JSONObject();
+        board.put("id", getBoardId());
+        final var contentWorkflowAction = new JSONObject();
+        contentWorkflowAction.put("workflow_action", "save_draft");
+        final var jsonTags = new JSONObject();
+        final var tagItems = new JSONArray();
+        for (final var tag : getTags()) {
+            tagItems.put(new JSONObject().put("text", tag));
+        }
+        jsonTags.put("items", tagItems);
+        final var data = new JSONObject();
+        data.put("type", "message");
+        data.put("board", board);
+        data.put("subject", getHeader());
+        data.put("body", getBody());
+        data.put("teaser", getTeaser());
```

Comment:
I think this one is evil since it makes `toJson()` non-deterministic. I would prefer if you have the teaser in a local variable here.

I think it you make testing easier if you select the teaser in a deterministic way. For example, take the hash of the release description modulo the number of available teasers. By that, the same release will always look the same but to users, it will look random.

Reply:
Sorry, I didn't get your idea. Can you please explain in other words? If you set the teaser yourself, then it's deterministic. If not, it's random. 
I don't want to over-engineer this little trick and count some hashes... If you want I can remove the random generation and we just provide teaser by hands for each release

Reply:
Maybe just decide based on the text length?   (`String teaser = TEASERS.get(getBody().size() % TEASERS.size())`)

Reply:
This is a good idea. I'll give it a try